src/main/java/dk/cngroup/lentils/service/ProgressService.java

```diff
@@ -74,4 +81,51 @@ public boolean isGameStartedForAllTeams() {
         }
         return true;
     }
+
+    public StageRangeOfTeams getCurrentStageRangeOfAllTeams() {
+        List<Team> teams = teamService.getAll();
+        List<Team> teamsWithCypherPending = getTeamsWithPendingCypher(teams);
+        List<Integer> stages = new ArrayList<>();
+        teamsWithCypherPending.stream()
+                .forEach(team -> stages.add(getStageOfPendingCypherForTeam(team)));
+        StageRangeOfTeams stageRangeOfTeams = new StageRangeOfTeams(Collections.min(stages),
+                Collections.max(stages));
+        return stageRangeOfTeams;
+    }
+
+    public int getNumberOfFinishedTeams() {
+        List<Team> teams = teamService.getAll();
+        int numberOfFinishedTeams = 0;
+        for (Team team : teams) {
+            if (gameLogicService.passedAllCyphers(team)) {
+                numberOfFinishedTeams++;
+            }
+        }
+        return numberOfFinishedTeams;
+    }
+
+    private List<Team> getTeamsWithPendingCypher(final List<Team> teams) {
+        List<Team> teamsWithCypherPending = teams.stream()
+                .filter(team -> hasCypherPending(team))
+                .collect(Collectors.toList());
+        return teamsWithCypherPending;
+    }
+
+    private boolean hasCypherPending(final Team team) {
+        return statusService.getAllByTeam(team)
+                .stream()
+                .filter(status -> status.getCypherStatus() == CypherStatus.PENDING)
+                .findAny()
+                .isPresent();
+    }
+
+    private int getStageOfPendingCypherForTeam(final Team team) {
+        List<Status> statuses = statusService.getAllByTeam(team);
+        Status statusPending = statuses
+                .stream()
```

Comment:
OPTIONAL: `.stream()` muzes nechat na predchozim radku, protoze se to tak vetsinou pise ;)